internal/controllers/synthesis/lifecycle.go

```diff
@@ -357,3 +346,32 @@ func shouldUpdateDeletedCompositionStatus(comp *apiv1.Composition) bool {
 func isReconciling(comp *apiv1.Composition) bool {
 	return comp.Status.CurrentSynthesis != nil && (comp.Status.CurrentSynthesis.Reconciled == nil || comp.Status.CurrentSynthesis.ObservedCompositionGeneration != comp.Generation)
 }
+
+// inputRevisionsEqual compares two sets of input revisions while ignoring deferred values.
+func inputRevisionsEqual(synth *apiv1.Synthesizer, a, b []apiv1.InputRevisions) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	refsByKey := map[string]apiv1.Ref{}
+	for _, ref := range synth.Spec.Refs {
+		ref := ref
+		refsByKey[ref.Key] = ref
+	}
+
+	var equal int
+	for _, a := range a {
```

Comment:
Mega nit, but maybe name the arguments ar/br and the loop variables a/b? Shadowing sometimes makes his hard to read, but it's just my preference so feel free to ignore.